cognite_toolkit/cdf_tk/load.py

```diff
@@ -60,249 +74,119 @@ def __next__(self):
-                    files.append(f)
-    # Read timeseries metadata
-    timeseries = TimeSeriesList([])
-    for f in files:
-        timeseries.extend(
-            TimeSeriesList.load(
-                load_yaml_inject_variables(Path(f"{directory}/{f}"), ToolGlobals.environment_variables())
-            ),
-        )
-    if len(timeseries) == 0:
-        return
-    print(f"[bold]Uploading {len(timeseries)} timeseries to CDF...[/]")
-    drop_ts: list[str] = []
-    for t in timeseries:
-        # Set the context info for this CDF project
-        t.data_set_id = ToolGlobals.data_set_id
-        if drop:
-            drop_ts.append(t.external_id)
-    try:
-        if drop:
-            if not dry_run:
-                client.time_series.delete(external_id=drop_ts, ignore_unknown_ids=True)
-                print(f"  Deleted {len(drop_ts)} timeseries.")
-            else:
-                print(f"  Would have deleted {len(drop_ts)} timeseries.")
-    except Exception:
-        print(f"[bold red]ERROR:[/] Failed to delete {t.external_id}. It may not exist.")
-    try:
-        if not dry_run:
-            client.time_series.create(timeseries)
+    load_files_individually: bool = False
+    filetypes = frozenset({"yaml", "yml"})
+    api_name: str
+    folder_name: str
+    resource_cls: type[CogniteResource]
+    list_cls: type[CogniteResourceList]
+
+    def __init__(self, client: CogniteClient):
+        self.client = client
+        self.api_class = self._get_api_class(client, self.api_name)
+
+    @staticmethod
+    def _get_api_class(client, api_name: str):
+        parent = client
+        if (dot_count := Counter(api_name)["."]) == 1:
+            parent_name, api_name = api_name.split(".")
+            parent = getattr(client, parent_name)
+        elif dot_count == 0:
+            pass
         else:
-            print(f"  Would have created {len(timeseries)} timeseries.")
-    except Exception as e:
-        print("[bold red]ERROR:[/] Failed to upload timeseries.")
-        print(e)
-        ToolGlobals.failed = True
-        return
-    print(f"  Created {len(timeseries)} timeseries from {len(files)} files.")
+            raise ValueError(f"Invalid api_name {api_name}.")
+        return getattr(parent, api_name)
+
+    @classmethod
+    def create_loader(cls, ToolGlobals: CDFToolConfig):
+        client = ToolGlobals.verify_capabilities(capability=cls.get_required_capability(ToolGlobals))
+        return cls(client)
+
+    @classmethod
+    @abstractmethod
+    def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
+        raise NotImplementedError(f"get_required_capability must be implemented for {cls.__name__}.")
+
+    @classmethod
+    @abstractmethod
+    def get_id(cls, item: T_Resource) -> T_ID:
+        raise NotImplementedError
+
+    # Default implementations that can be overridden
+    def create(
+        self, items: T_Resource | Sequence[T_Resource], ToolGlobals: CDFToolConfig, drop: bool
```

Comment:
Have you ditched the drop functionality? We have --drop and --drop-data. I think we should introduce a diffing as default create behaviour, i.e. we should retrieve the yaml and compare and upsert where we can. However, having a --drop flag that will actively delete the resources before creating again, is a needed functionality. E.g. where you want to clean out transformation job history or be absolutely sure that you are starting from scratch. I think we should ditch --drop-data now and rely on cdf-tk clean only.